### DISCUSSION TITLE
State hooks: Very coupled and presenting unexpected behaviors

### DISCUSSION THREAD

**Discussion by davicostalf** [vtex/shoreline#738] [General]
Comentários no CSAT de Engenharia em 2023 H1:

> Some state hooks don't compose well with patterns such as suspense, virtualization, and laziness.

> O que atrapalha no debug geralmente é quando é algum comportamento inesperado dentro dos hooks de state, pois parte do código deles está no admin ui mas parte vem de bibliotecas externas, então requer um tempo maior de análise pra entender como funcionam.

> hooks de estado que são acoplados demais

**Comment by matheusps** [comment]
On Shoreline it was resolved by making all the components controllable. You can still use hooks to manage the component inner store, but it's optional. https://github.com/vtex/shoreline/discussions/1025.
